Pass version as command line to controller

nhaghighat opened the pull request:
## Description

This PR passes the default version as a command line argument so that the controller rebuild would be fast.

Fixes: #989

## Checklist

- [x] I have manually tested my changes and I am happy with the result.
- [x] The documentation reflects the changes.
- [x] The PR branch is set up to merge into `dev` instead of `master`.

yannickperrenet commented on services/orchest-controller/Makefile:
```diff
@@ -25,7 +25,7 @@ fmt: ## Run go fmt and modify files in place
 .PHONY: controller
 controller: fmt
-	CGO_ENABLED=0 GOOS=linux go build -ldflags="$(LDFLAGS)" -o bin/controller ./cmd/controller
+	CGO_ENABLED=0 GOOS=linux go build -o bin/controller ./cmd/controller
```
Can't we now also remove the `LDFLAGS` variable definition?

yannickperrenet commented:
When I try to build Orchest using:
```sh
export TAG="$(orchest version --latest)"
scripts/build_container.sh -M -t $TAG -o $TAG
```
I get:
```
make: Entering directory '/home/yannick/Documents/Orchest/orchest/services/orchest-controller'
Updating CRD manifests
/home/yannick/Documents/Orchest/orchest/.cache/tools/controller-gen crd paths=./pkg/apis/... output:dir=deploy/helm/templates
Generating orchest-controller.yaml manifests
Error: parse error at (orchest-controller/templates/deployment.yaml:26): function "Values" not defined

Use --debug flag to render out invalid YAML
make: *** [Makefile:59: manifestgen] Error 1
make: Leaving directory '/home/yannick/Documents/Orchest/orchest/services/orchest-controller'
```

yannickperrenet commented:
Given that this PR sets out to solve https://github.com/orchest/orchest/issues/989, it should also pick up the linked discussions in the issue description. Otherwise we lose track of what has been fixed and what hasn't been.

nhaghighat commented:
> Given that this PR sets out to solve https://github.com/orchest/orchest/issues/989, it should also pick up the linked discussions in the issue description. Otherwise, we lose track of what has been fixed and what hasn't been.

addressed the comment in the mentioned issue.

> From what I understand this PR sets out to make this rebuild fast

The problem was that `make manifestgen` was running before build and was changing the filesystem, added k8s generated folder to the `.dockerignore` to resolve the issue
